### ISSUE TITLE
CKV_AWS_45 False Positive

### ISSUE THREAD

**Issue by bakosa** [bridgecrewio/checkov#5743]
**Describe the issue**
[CKV_AWS_45](https://github.com/bridgecrewio/checkov/blob/main/checkov/cloudformation/checks/resource/aws/LambdaEnvironmentCredentials.py) gives a false positive to any value after a lambda environment variable that is exactly 40 characters long due to the AWS secret access key [regex](https://github.com/bridgecrewio/checkov/blob/4875adc472f409bc7039a879347b404468e88215/checkov/common/util/secrets.py#L49). Obviously this regex makes sense. However, this check could do a deeper analysis on the environment variable being passed in to have less false positives. In the example below, I think it would be at least a good idea to check if an intrinsic function is used (in this case `!Ref`) in order to reduce false positives. 

**Examples**
```yml
Resources:
  MyCoolLambdaFunction:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: blah
      Handler: src.app.lambda_handler
      Policies:
      - StepFunctionsExecutionPolicy:
          StateMachineName: !GetAtt MySuperCoolFortyCharLongStateMachineeeee.Name
      Environment:
        Variables:
          MY_COOL_STATE_MACHINE: !Ref MySuperCoolFortyCharLongStateMachineeeee # 40 chars in ref fails check
```
                                                                    
 
**Version (please complete the following information):**
 - Checkov Version latest

**Additional context**

You can see how this works by calling the `get_secrets_from_string` method directly in a python terminal which is what is being called in the `CKV_AWS_45` [check](https://github.com/bridgecrewio/checkov/blob/4875adc472f409bc7039a879347b404468e88215/checkov/cloudformation/checks/resource/aws/LambdaEnvironmentCredentials.py#L27).

```python
from checkov.common.util.secrets import AWS, GENERAL, get_secrets_from_string

get_secrets_from_string("MySuperCoolFortyCharLongStateMachineeeee", AWS, GENERAL)
# ['MySuperCoolFortyCharLongStateMachineeeee']

get_secrets_from_string("x"*40, AWS, GENERAL)
# ['xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx']

get_secrets_from_string("x"*41, AWS, GENERAL)
# []
```

**Comment by gruebel** [comment]
hey @bakosa thanks for reaching out.

Yeah, seems like a very good idea 😄 I will create a PR for it

**Comment by bakosa** [comment]
Thank you!!
